test/jdk/java/foreign/StdLibTest.java

```diff
@@ -347,14 +349,9 @@ private MethodHandle specializedPrintf(List<PrintfArg> args) {
             for (PrintfArg arg : args) {
                 mt = mt.appendParameterTypes(arg.carrier);
             }
-            //function
-            List<MemoryLayout> argLayouts = new ArrayList<>();
-            argLayouts.add(C_POINTER); //format
-            for (PrintfArg arg : args) {
-                argLayouts.add(arg.layout);
-            }
-            MethodHandle mh = abi.downcallHandle(printfAddr, mt,
-                    FunctionDescriptor.of(C_INT, false, argLayouts.toArray(new MemoryLayout[0])));
+            FunctionDescriptor printfSpec = StdLibHelper.printfBase.appendArgumentLayouts(
```

Comment:
I would have dealt with expanding the function descriptor in the same loop where we expand the method type - but what you have is also fine.

Reply:
Yeah, I like that better as well. I'll change it.

I guess I got a little tunnel visioned since the comment pointed at the line with `FunctionDescriptor.of(C_INT, argLayouts.toArray(new MemoryLayout[0])))` on GitHub, and doesn't show the for loop where the MethodType is expanded in the first place (neither does the email it looks like). From what I understand, multi-line review comments are in the works, so in the future it should be possible to attach a comment to a larger block of code.